### PR TITLE
Feature/api detail page

### PR DESCRIPTION
**Pull request by itsnameissteven**
## What does this PR do?
1. Detail page has been built out 
2. notes can be saved for a specific API
3. API can be favorited
4. Data persists via localStorage
-

## How should it be tested in the UI?
click on an api card to go to the detail page 
write and save a note, the note should render on the right of page.
save an api as a favorite. this will not show up anywhere but can be found in local storage

-

### CODE REVIEW

File: src/components/App/App.tsx [hunk 1]
```diff
@@ -45,39 +57,78 @@ class App extends React.Component<Props> {
...
+        }
+        return savedNote;
+      });
+      return this.setState({ savedNotes: updatedNotes});
+    } 
+    this.setState({ savedNotes: [...this.state.savedNotes, {name: ApiName, notes: [note]}]});
   }
 
   render() {
+    this.state.favorites.length && localStorage.setItem('favorites', JSON.stringify(this.state.favorites));
```
Comment (Jacksonmcguire) [inline src/components/App/App.tsx:102]: Good call with this render conditional to keep local storage clean

File: src/components/FeaturedCard/FeaturedCard.tsx [hunk 2]
```diff
@@ -1,16 +1,67 @@
...
+  handleChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    this.setState({note: event.target.value})
+  }
+
+  saveCurrentNote = () => {
+    this.props.saveNote(this.state.API, this.state.note);
+    this.setState({note: ''});
+  }
+
+  render() {
```
Comment (Jacksonmcguire) [inline src/components/FeaturedCard/FeaturedCard.tsx:35]: Sweet, featured card's render is looking way better

**Review by Jacksonmcguire — APPROVED** [review]
Didn't have much to say, everything is looking super clean to me. Good stuff!

File: src/components/FeaturedCard/FeaturedCard.tsx [hunk 3]
```diff
@@ -1,16 +1,67 @@
...
+      <div className="featured-card">
+        <h2>{this.props.API}</h2>
+        <p>Authentication: {this.props.Auth}</p>
+        <p>Cors? {this.props.Cors}</p>
+        <p>{this.props.HTTPS}</p>
+        <p>Category: {this.props.Category}</p>
+        <p>{this.props.Description}</p>
+        <p>
+          <a href={this.props.Link} target="_blank">
+            more details
```
Comment (cbmackintosh) [inline src/components/FeaturedCard/FeaturedCard.tsx:46]: Getting a warning on my console about this target property on line 45 - Using target="_blank" without rel="noreferrer" is a security risk. Not sure what that's about, but something to keep in mind.

File: src/components/FeaturedCard/FeaturedCard.tsx [hunk 4]
```diff
@@ -1,16 +1,67 @@
...
+          value={this.state.note}
+          className="featured-card__notes"
+          placeholder="Notes"
+          onChange={this.handleChange}
+        />
+        <button onClick={this.saveCurrentNote}>Save Notes</button>
+        <button
+          onClick={() =>
+            this.state.API && this.props.addToFavorites(this.state)
+          }
```
Comment (cbmackintosh) [inline src/components/FeaturedCard/FeaturedCard.tsx:60]: Verified that the data is saving as expected to localStorage. No problems with duplicate entries either. I wonder if we can change the 'Add to Favorites' button to some kind of toggle so there's some indication that the card has been saved. Is there a way to un-favorite?

**Review by cbmackintosh — APPROVED** [review]
Everything working smoothly - good progress on this feature!

**Pull request merged**